src/features/calendar/components/CalendarEventFIlter/EventFilterPane.tsx

```diff
@@ -0,0 +1,299 @@
+        ],
+      })
+    );
+  };
+
+  const resetFilters = () => {
+    const filterCategories: FilterCategoryType[] = [
+      'selectedActions',
+      'selectedStates',
+      'selectedTypes',
+    ];
+    filterCategories.map((filterCategory) =>
+      store.dispatch(
+        filterUpdated({
+          filterCategory: filterCategory,
+          selectedFilterValue: [],
+        })
+      )
+    );
+  };
+
+  const debouncedFinishedTyping = useDebounce(async (value: string) => {
+    store.dispatch(filterTextAdded({ filterText: value }));
+  }, 400);
+
+  return (
+    <>
+      <PaneHeader title={messages.eventFilter.filter()} />
+      <Button
+        color="warning"
+        onClick={resetFilters}
+        size="small"
+        variant="outlined"
+      >
+        <Msg id={messageIds.eventFilter.reset} />
+      </Button>
+      <Box sx={{ mt: 2 }}>
+        <TextField
+          fullWidth
+          InputProps={{
+            endAdornment: (
+              <>
+                {userInput && (
+                  <IconButton
+                    onClick={() => {
+                      setUserInput('');
+                      debouncedFinishedTyping('');
+                    }}
+                  >
+                    <Clear />
+                  </IconButton>
+                )}
+              </>
+            ),
+            startAdornment: (
+              <InputAdornment position="start">
+                <FilterList />
+              </InputAdornment>
+            ),
+          }}
+          onChange={(e) => {
+            setUserInput(e.target.value);
+            debouncedFinishedTyping(e.target.value);
+          }}
+          placeholder={messages.eventFilter.type()}
+          value={userInput}
+        />
+        <Box display="flex" flexDirection="column" sx={{ mt: 2 }}>
+          <FormGroup sx={{ mb: 2 }}>
+            <Box
+              alignItems="center"
+              display="flex"
+              justifyContent="space-between"
+            >
+              <Typography color="secondary" variant="body1">
+                <Msg
+                  id={messageIds.eventFilter.filterOptions.actionFilters.title}
+                />
+              </Typography>
+              <AllAndNoneToggle
+                filterCategory="selectedActions"
+                selectedFilterLength={state.selectedActions.length}
+              />
+            </Box>
+            <>
+              {Object.entries(ACTION_FILTER_OPTIONS).map((options) => {
+                const enumKey = options[0];
+                const enumValue = options[1];
+
+                const entries = Object.entries(
+                  messages.eventFilter.filterOptions.actionFilters
+                );
+                const lowerCaseKey = enumKey.toLowerCase();
+                const message = entries.find(
+                  (entry) => entry[0] === lowerCaseKey
+                );
+
+                if (!message) {
+                  return null;
+                }
```

Comment:
You've made this more complicated than it needs to be. Let's just save the correct keys for messages in the enum, instead of human readable values.